### PR TITLE
fix for PLUG-224,property name space validation

### PR DESCRIPTION
**Pull request by vitiwari**
@gokumar pull request contains following.
1) validation for properties field in event
2) Plugin should not throw exception if no instance configured
3) single line event definitions bug fix

### CODE REVIEW

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/RemedyPlugin.java [hunk 1]
```diff
@@ -80,44 +80,47 @@ public void setDispatcher(CollectorDispatcher dispatcher) {
...
+                TemplateValidator templateValidator = new PluginTemplateValidator();
+                try {
+                    templateValidator.validate(template);
+
+                } catch (ValidationException ex) {
+                    System.err.println("Validation failed - " + ex.getMessage());
+                    eventOutput.emit(Util.eventMeterTSI(Constants.REMEDY_PLUGIN_TITLE_MSG, ex.getMessage(), Event.EventSeverity.ERROR.toString()));
+                    isValidJson = false;
+                }
+                if (isValidJson) {
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/meter/plugin/remedy/RemedyPlugin.java]: nitty gritty..

when you are talking about Template everywhere, call it isValidTemplate, what do you think?

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java [hunk 2]
```diff
@@ -45,7 +45,9 @@ public boolean validate(Template template) throws ValidationException {
         // validate payload configuration
         Map<String, String> properties = payload.getProperties();
         for (String key : properties.keySet()) {
-
+        	if(!StringUtil.isValidJavaIdentifier(key)){
+        		throw new ValidationException(StringUtil.format(Constants.PROPERTY_NAME_CONTAINS_SPACE, new Object[]{key.trim()}));
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java]: now that it is  not just space, should we call it something else?

`Constants.PROPERTY_NAME_CONTAINS_SPACE`

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/Util.java [hunk 3]
```diff
@@ -37,7 +37,7 @@ public static String dateToString(Date date) {
 
     public static String getFieldValues(String fieldValue[]) {
         StringBuffer fieldValues = new StringBuffer();
-        if (fieldValue != null && fieldValue.length > 1) {
+        if (fieldValue != null && fieldValue.length > 0) {
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/meter/plugin/remedy/util/Util.java:40]: was this a bug earlier why was it >1?

**Review by gokumar — APPROVED** [review]

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/Util.java (hunk 3 above)
Comment (vitiwari) [reply src/main/java/com/bmc/truesight/meter/plugin/remedy/util/Util.java:40]: Yes it was a bug, this was written by Santosh. I missed it.

**Review by gokumar — APPROVED** [review]

**Pull request merged**